jwst/pathloss/tests/test_pathloss.py

```diff
@@ -0,0 +1,332 @@
+    datmod = MultiSlitModel()
+    x_pos, y_pos = get_center("NRC_IMAGE", datmod)
+
+    assert(x_pos==y_pos==0.0)
+
+
+def test_get_center_exptype():
+    """ If exptype is "NRS_MSASPEC" | "NRS_FIXEDSLIT" | "NRS_BRIGHTOBJ" and
+    source_xpos and source_ypos exist in datamod.slits, make sure it's returned
+    """
+    
+    datmod = MultiSlitModel()
+    datmod.slits.append({'source_xpos':1, 'source_ypos':2})
+    
+    for exptype in ["NRS_MSASPEC", "NRS_FIXEDSLIT", "NRS_BRIGHTOBJ"]:
+        x_pos, y_pos = get_center(exptype, datmod.slits[0])
+
+        assert(x_pos==1)
+        assert(y_pos==2)
+
+
+# Begin get_aperture_from_model tests
+def test_get_app_from_model_null():
+    """If exp_type isn't the NRS or NIS specific mode,
+    routine returns None
+    """
+
+    datmod = MultiSlitModel()
+    datmod.meta.exposure.type = 'NRC_IMAGE'
+
+    result = get_aperture_from_model(datmod, None)
+
+    assert(result==None)
+
+
+def test_get_aper_from_model_fixedslit():
+    """For a given exposures aperture, make sure the correct
+    aperture reference data is returned for fixedslit mode
+    """
+
+    datmod = PathlossModel()
+    datmod.apertures.append({'name':'S200A1'})
+    datmod.meta.exposure.type = 'NRS_FIXEDSLIT'
+
+    result = get_aperture_from_model(datmod, 'S200A1')
+
+    assert(result == datmod.apertures[0])
+
+
+def test_get_aper_from_model_msa():
+    """For a given exposures aperture, make sure the correct
+    aperture reference data is returned for MSA mode
+    """
+
+    datmod = PathlossModel()
+    datmod.apertures.append({'shutters':1})
+    datmod.meta.exposure.type = 'NRS_MSASPEC'
+
+    result = get_aperture_from_model(datmod, 1)
+
+    assert(result == datmod.apertures[0])
+
+
+# Begin calculate_pathloss_vector tests.
+def test_calculate_pathloss_vector_pointsource_data():
+    """Calculate pathloss vector for 3D pathloss data
+    """
+    datmod = PathlossModel()
+
+    ref_data = {'pointsource_data':np.ones((10,10,10), dtype=np.float32),
+                'pointsource_wcs': {'crval2': -0.5, 'crpix2': 1.0, 'cdelt2': 0.05, 
+                                    'cdelt3': 1, 'crval1': -0.5, 'crpix1': 1.0, 
+                                    'crpix3': 1.0, 'crval3': 1, 'cdelt1': 0.05}                  
+                }
+
+    datmod.apertures.append(ref_data)
+
+    wavelength, pathloss, is_inside_slitlet = calculate_pathloss_vector(datmod.apertures[0].pointsource_data,
+                                                                        datmod.apertures[0].pointsource_wcs,
+                                                                        0.0, 0.0)
+
+    # Wavelength array is calculated with this: crval3 +(float(i+1) - crpix3)*cdelt3
+    # Where i is the iteration of np.arange(wavesize) which is the 1st dimension of the pointsource
+    # data array.
+    wavelength_comparison = np.array([1 + (float(i+1) - 1.0)*1 for i in np.arange(10)])
+    assert(np.all(wavelength==wavelength_comparison))
```

Comment:
For float data it might be better to use `np.allclose(wavelength, wavelength_comparison)` instead of `np.all(wavelength == wavelength_comparison)`.  `np.allclose` lets you specify absolute and/or relative tolerances for the comparison.

Reply:
Good idea, I went ahead and made the change.